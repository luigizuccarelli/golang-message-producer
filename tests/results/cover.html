
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">/home/lzuccarelli/Projects/myportfolio/myportfolio-message-producer/handlers.go (100.0%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">not covered</span>
				<span class="cov8">covered</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">package main

import (
        "encoding/json"
        "fmt"
        "io/ioutil"
        "net/http"
        "os"
        "strings"
)

const (
        CONTENTTYPE     string = "Content-Type"
        APPLICATIONJSON string = "application/json"
)

// Streamer a http response and request for a message producer
// @param - http.ResponseWriter (used to send back response)
// @param - http.Request object
func Streamer(w http.ResponseWriter, r *http.Request) <span class="cov8" title="1">{

        var response Response

        addHeaders(w, r)
        handleOptions(w, r)

        body, err := ioutil.ReadAll(r.Body)
        if err != nil </span><span class="cov8" title="1">{
                response = Response{StatusCode: "500", Status: "ERROR", Message: "Could not read body data " + err.Error()}
                w.WriteHeader(http.StatusInternalServerError)
        }</span> else<span class="cov8" title="1"> {
                err = connectors.SendMessageSync(body)
                if err != nil </span><span class="cov8" title="1">{
                        response = Response{StatusCode: "500", Status: "ERROR", Message: "Could not send stream data " + err.Error()}
                        w.WriteHeader(http.StatusInternalServerError)
                }</span> else<span class="cov8" title="1"> {
                        response = Response{StatusCode: "200", Status: "OK", Message: "Stream data sent successfully"}
                }</span>
        }

        <span class="cov8" title="1">b, _ := json.MarshalIndent(response, "", "        ")
        fmt.Fprintf(w, string(b))</span>
}

// IsAlive a http response and request wrapper for health endpoint checks
// It takes a both response and request objects and returns void
func IsAlive(w http.ResponseWriter, r *http.Request) <span class="cov8" title="1">{
        addHeaders(w, r)
        handleOptions(w, r)
        logger.Trace(fmt.Sprintf("used to mask cc %v", r))
        fmt.Fprintf(w, "{\"version\": \""+os.Getenv("VERSION")+"\"}")
}</span>

// headers (with cors) utility
func addHeaders(w http.ResponseWriter, r *http.Request) <span class="cov8" title="1">{
        var request []string
        for name, headers := range r.Header </span><span class="cov8" title="1">{
                name = strings.ToLower(name)
                for _, h := range headers </span><span class="cov8" title="1">{
                        request = append(request, fmt.Sprintf("%v: %v", name, h))
                }</span>
        }

        <span class="cov8" title="1">logger.Trace(fmt.Sprintf("Headers : %s", request))

        w.Header().Set(CONTENTTYPE, APPLICATIONJSON)
        // use this for cors
        w.Header().Set("Access-Control-Allow-Origin", "*")
        w.Header().Set("Access-Control-Allow-Methods", "POST, GET, OPTIONS, PUT, DELETE")
        w.Header().Set("Access-Control-Allow-Headers", "Accept, Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization")</span>
}

// simple options handler
func handleOptions(w http.ResponseWriter, r *http.Request) <span class="cov8" title="1">{
        if r.Method == "OPTIONS" </span><span class="cov8" title="1">{
                w.WriteHeader(http.StatusOK)
                fmt.Fprintf(w, "")
        }</span>
        <span class="cov8" title="1">return</span>
}
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
