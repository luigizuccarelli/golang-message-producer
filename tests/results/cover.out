mode: set
/home/lzuccarelli/Projects/myportfolio/myportfolio-message-producer/handlers.go:20.55,28.16 5 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-message-producer/handlers.go:41.2,42.27 2 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-message-producer/handlers.go:28.16,31.3 2 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-message-producer/handlers.go:31.8,33.17 2 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-message-producer/handlers.go:33.17,36.4 2 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-message-producer/handlers.go:36.9,38.4 1 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-message-producer/handlers.go:47.54,52.2 4 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-message-producer/handlers.go:55.57,57.38 2 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-message-producer/handlers.go:64.2,70.134 5 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-message-producer/handlers.go:57.38,59.29 2 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-message-producer/handlers.go:59.29,61.4 1 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-message-producer/handlers.go:74.60,75.27 1 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-message-producer/handlers.go:79.2,79.8 1 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-message-producer/handlers.go:75.27,78.3 2 1
